src/sentry/api/event_search.py

```diff
@@ -624,11 +627,27 @@ def get_snuba_query_args(query=None, params=None):
         'filter_keys': defaultdict(list),
     }
 
+    projects = {}
+    has_project_term = any(
+        isinstance(term, SearchFilter) and term.key.name == PROJECT_KEY
+        for term
+        in parsed_terms
+    )
+    if has_project_term:
+        projects = {
+            p['slug']: p['id'] for p in Project.objects.filter(
+                id__in=params['project_id'],
+            ).values('id', 'slug')
+        }
+
     for term in parsed_terms:
         if isinstance(term, SearchFilter):
             snuba_name = term.key.snuba_name
+            if term.key.name == PROJECT_KEY:
+                condition = ['project_id', '=', projects.get(term.value.value)]
```

Comment:
Does `=` do the right thing when there are multiple projectids? If `term.value.value` is a single project why do you need to use `in` when fetching projectslug:id mapping?

Reply:
I suppose we only really support one right now, but I wanted to use the `in` to support OR search in future